### PR TITLE
 YSQL Regression Tests for Join Clause

### PR DESCRIPTION
**Pull request by ngov17**
This pull request adds a test for join (`yb_pg_join`) from the  postgres framework `src/postgres/src/test/regress/`

This pull request was originally part of the larger pull request https://github.com/yugabyte/yugabyte-db/pull/5190 (now edited) which has now been broken down into 4 separate pull requests:  https://github.com/yugabyte/yugabyte-db/pull/5190 (edited), https://github.com/yugabyte/yugabyte-db/pull/5262, https://github.com/yugabyte/yugabyte-db/pull/5263, https://github.com/yugabyte/yugabyte-db/pull/5264.

### CODE REVIEW

File: src/postgres/src/test/regress/yb_pg_random_schedule [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+# This includes tests for Random Functions
+####################################################################################################
+
+# Chain of dependencies for `yb_pg_random`
+test: yb_pg_create_function_1
+test: yb_pg_create_type
+test: yb_pg_create_table
+test: yb_pg_copy
+
+test: random
```
Comment (jaki) [inline src/postgres/src/test/regress/yb_pg_random_schedule:13]: nit: missing newline at EOF (ditto elsewhere)

File: java/yb-pgsql/src/test/java/org/yb/pgsql/TestPgDelete.java [hunk 2]
```diff
@@ -201,4 +201,47 @@ public void testDeleteReturn() throws SQLException {
...
+    createSimpleTable(tableName1);
+    createSimpleTable(tableName2);
+
+    //Fill in the helper table:
+     try (Statement insert_stmt = connection.createStatement()) {
+      insert_stmt.execute("INSERT INTO test_helper(h, r, vi, vs) VALUES(1, 0.5, 10, 'v')");
+     }
+    
+
+
```
Comment (jaki) [inline java/yb-pgsql/src/test/java/org/yb/pgsql/TestPgDelete.java:219]: nit: too many newlines (ditto elsewhere)

File: java/yb-pgsql/src/test/java/org/yb/pgsql/TestPgDelete.java [hunk 3]
```diff
@@ -201,4 +201,47 @@ public void testDeleteReturn() throws SQLException {
...
+      String insert_format = "INSERT INTO %s(h, r, vi, vs) VALUES(%d, %f, %d, '%s')";
+      for (long h = 0; h < 5; h++) {
+          String insert_text = String.format(insert_format, tableName1,
+                                             h, h + 0.5, h * 10 + h, "v" + h );
+          if (h == 1) {
+            // Constructing rows to be returned by DELETE.
+            expectedRows.add(new Row(h, h + 0.5 , h * 10 + h, "v" + h, 1, 0.5, 10, "v"));
+          }
+          insert_stmt.execute(insert_text);
+
```
Comment (jaki) [inline java/yb-pgsql/src/test/java/org/yb/pgsql/TestPgDelete.java:231]: nit: no need for newline here

File: src/postgres/src/test/regress/yb_aggregates_schedule [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+# Dependencies for `yb_pg_aggregates`
+test: yb_pg_int4
+test: yb_pg_int8
+test: yb_pg_create_aggregate
+
+# Chain of dependencies for `yb_pg_aggregates`
+test: yb_pg_create_function_1
+test: yb_pg_create_type
+test: yb_pg_create_table
+test: yb_pg_copy
```
Comment (jaki) [inline src/postgres/src/test/regress/yb_aggregates_schedule:15]: Please move `yb_pg_create_aggregate` back out to a separate section as in `yb_pg_misc_serial_schedule`.  The ordering should align with `serial_schedule`'s ordering because it has the least chance of chain dependency issues for now and the future.

File: src/postgres/src/test/regress/yb_pg_groupingsets_schedule [hunk 5]
```diff
@@ -0,0 +1,15 @@
+# src/test/regress/yb_pg_groupingsets_schedule
+#
+####################################################################################################
+# This includes tests for Group
+####################################################################################################
+# Dependencies for `yb_pg_groupingsets`
+test: yb_pg_int8
+
+# Chain of dependencies for `yb_pg_window`
```
Comment (jaki) [inline src/postgres/src/test/regress/yb_pg_groupingsets_schedule:9]: Looks like a copy/paste mistake.  Can you make sure that `yb_pg_copy` is really needed for `yb_pg_groupingsets`?

File: src/postgres/src/test/regress/yb_pg_join_schedule [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+test: yb_pg_int8
+test: yb_float8
+test: yb_text
+
+
+# Chain of dependencies for `yb_pg_join`
+test: yb_pg_create_function_1
+test: yb_pg_create_type
+test: yb_pg_create_table
+test: yb_pg_copy
```
Comment (jaki) [inline src/postgres/src/test/regress/yb_pg_join_schedule:18]: Again, can you verify this dependency is valid?  Ditto for the rest of the schedules.

File: src/postgres/src/test/regress/yb_pg_join_schedule [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+#
+####################################################################################################
+# This includes tests for Join Clause
+####################################################################################################
+# Dependencies for `yb_pg_join`
+test: yb_pg_int4
+test: yb_int2
+test: yb_pg_int8
+test: yb_float8
+test: yb_text
```
Comment (jaki) [inline src/postgres/src/test/regress/yb_pg_join_schedule:11]: I don't understand why `yb_foo` tests are dependencies for `yb_pg_join`.  Can you explain?

File: src/postgres/src/test/regress/yb_pg_limit_schedule [hunk 8]
```diff
@@ -0,0 +1,16 @@
+# src/test/regress/yb_pg_limit_schedule
+#
+####################################################################################################
+# This includes tests for LIMIT Clause
+####################################################################################################
+# Dependencies for `yb_pg_limit`
+test: yb_pg_int8
+
```
Comment (jaki) [inline src/postgres/src/test/regress/yb_pg_limit_schedule:8]: nit: remove extra newline

File: java/yb-pgsql/src/test/java/org/yb/pgsql/TestPgDelete.java [hunk 9]
```diff
@@ -201,4 +201,47 @@ public void testDeleteReturn() throws SQLException {
...
+
+  @Test
+  public void testDeleteUsing() throws SQLException {
+    String tableName1 = "test_delete_using";
+    String tableName2 = "test_helper";
+    createSimpleTable(tableName1);
+    createSimpleTable(tableName2);
+
+    //Fill in the helper table:
```
Comment (jaki) [inline java/yb-pgsql/src/test/java/org/yb/pgsql/TestPgDelete.java:213]: nit: space after `//`

File: src/postgres/src/test/regress/yb_pg_random_schedule (hunk 1 above)
Comment (jaki) [inline src/postgres/src/test/regress/yb_pg_random_schedule:13]: Can you port `random` to `yb_pg_random`?  Also, it seems like overkill to create a schedule for effectively a single test: can you move this to `yb_pg_misc_serial_schedule`?

File: src/postgres/src/test/regress/yb_pg_limit_schedule [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+
+# Chain of dependencies for `yb_pg_limit`
+test: yb_pg_create_function_1
+test: yb_pg_create_type
+test: yb_pg_create_table
+test: yb_pg_copy
+
+test: yb_pg_limit
```
Comment (jaki) [inline src/postgres/src/test/regress/yb_pg_limit_schedule:16]: Seems like overkill to create a schedule for effectively a single test: can you move this to `yb_pg_misc_serial_schedule` (unless it has no dependencies: then it goes to `yb_pg_misc_independent_serial_schedule`)?

File: src/postgres/src/test/regress/yb_pg_groupingsets_schedule [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+test: yb_pg_create_function_1
+test: yb_pg_create_type
+test: yb_pg_create_table
+test: yb_pg_copy
+
+test: yb_pg_groupingsets
```
Comment (jaki) [inline src/postgres/src/test/regress/yb_pg_groupingsets_schedule:15]: Seems like overkill to create a schedule for effectively a single test: can you move this to `yb_pg_misc_serial_schedule`?

File: src/postgres/src/test/regress/expected/yb_float8.out [hunk 12]
```diff
@@ -556,4 +556,3 @@ FROM (SELECT 10*cosd(a), 10*sind(a)
 (5 rows)
 
 RESET extra_float_digits;
-DROP TABLE FLOAT8_TBL;
```
Comment (jaki) [inline src/postgres/src/test/regress/expected/yb_float8.out:559]: Note: although this reduces diff with the original test, it might make our existing tests fail.  We should keep this change until some obvious error shows up like a test depending on that table existing.

File: src/postgres/src/test/regress/yb_pg_copy2_schedule [hunk 13]
```diff
@@ -0,0 +1,7 @@
+# src/test/regress/yb_pg_copy2_schedule
+#
+####################################################################################################
+# Tests for Copy 
+####################################################################################################
+
+test: yb_pg_copy2
```
[the remark this answers is not in the document]
Comment (ngov17) [reply src/postgres/src/test/regress/yb_pg_copy2_schedule]: Sure. However, `yb_pg_copy` is not exactly a test; it is used in a chain of dependencies
```
test: yb_pg_create_function_1
test: yb_pg_create_type
test: yb_pg_create_table
test: yb_pg_copy
```
 to actually fill and populate tables used in other tests like copy2. So, I'll create `yb_pg_copy_schedule` with `yb_pg_copy2` and `yb_pg_copydml`.

File: src/postgres/src/test/regress/yb_aggregates_schedule (hunk 4 above)
Comment (ngov17) [reply src/postgres/src/test/regress/yb_aggregates_schedule]: I've created `yb_pg_aggregates_schedule` that has the right ordering in the new pull request. Also, since `yb_pg_aggregates` will have its own schedule, I'll delete it in `yb_pg_misc_serial_schedule` when I add the other (effectively single) tests.

File: src/postgres/src/test/regress/yb_pg_join_schedule (hunk 6 above)
Comment (ngov17) [reply src/postgres/src/test/regress/yb_pg_join_schedule:18]: It is a chain of dependencies needed to fill the tables for the tests.

File: src/postgres/src/test/regress/yb_pg_join_schedule (hunk 7 above)
Comment (ngov17) [reply src/postgres/src/test/regress/yb_pg_join_schedule]: Even `yb_int2 `and` yb_text` turned out to be ported tests. I've renamed them too accordingly.

File: src/postgres/src/test/regress/yb_pg_groupingsets_schedule (hunk 11 above)
Comment (ngov17) [reply src/postgres/src/test/regress/yb_pg_groupingsets_schedule]: I've added` yb_pg_limit`,` yb_pg_random`, `yb_pg_groupingsets` to `yb_pg_misc_serial_schedule` in a separate pull request. In that, I've also removed the tests for `yb_pg_aggregates `as mentioned in https://github.com/yugabyte/yugabyte-db/pull/5190#discussion_r462831602

File: src/postgres/src/test/regress/yb_pg_groupingsets_schedule (hunk 5 above)
Comment (ngov17) [reply src/postgres/src/test/regress/yb_pg_groupingsets_schedule]: See: https://github.com/yugabyte/yugabyte-db/pull/5190#discussion_r462827939

**Comment by ngov17** [comment]
I divided this pull request into four parts:  https://github.com/yugabyte/yugabyte-db/pull/5190 (edited), https://github.com/yugabyte/yugabyte-db/pull/5262, https://github.com/yugabyte/yugabyte-db/pull/5263, https://github.com/yugabyte/yugabyte-db/pull/5264.

File: src/postgres/src/test/regress/yb_pg_copy2_schedule (hunk 13 above)
[the remark this answers is not in the document]
Comment (ngov17) [reply src/postgres/src/test/regress/yb_pg_copy2_schedule]: See: https://github.com/yugabyte/yugabyte-db/pull/5264

File: src/postgres/src/test/regress/yb_aggregates_schedule (hunk 4 above)
Comment (ngov17) [reply src/postgres/src/test/regress/yb_aggregates_schedule]: See: https://github.com/yugabyte/yugabyte-db/pull/5263

File: src/postgres/src/test/regress/yb_pg_groupingsets_schedule (hunk 11 above)
Comment (ngov17) [reply src/postgres/src/test/regress/yb_pg_groupingsets_schedule]: See: https://github.com/yugabyte/yugabyte-db/pull/5264

File: src/postgres/src/test/regress/expected/yb_pg_join.out [hunk 14]
```diff
@@ -0,0 +1,6598 @@
...
+ name | code | b_cnt | const
+------+------+-------+-------
+ A    | p    |     2 |    -1
+ B    | q    |     0 |    -1
+ C    |      |       |
+(3 rows)
+
+drop table a;
+drop table b;
+drop table c;
```
Comment (jaki) [inline src/postgres/src/test/regress/expected/yb_pg_join.out:2789]: Why were these added?

File: src/postgres/src/test/regress/expected/yb_pg_join.out [hunk 15]
```diff
@@ -0,0 +1,6598 @@
...
+set enable_nestloop = 0;
+create temp table a (i integer);
+create temp table b (x integer, y integer);
+select * from a left join b on i = x and i = y and x = i;
+ i | x | y
+---+---+---
+(0 rows)
+
+drop table a;
+drop table b;
```
Comment (jaki) [inline src/postgres/src/test/regress/expected/yb_pg_join.out:2666]: Why are these added?

File: src/postgres/src/test/regress/expected/yb_pg_join.out [hunk 16]
```diff
@@ -0,0 +1,6598 @@
...
+        QUERY PLAN
+--------------------------
+ Result
+   One-Time Filter: false
+(2 rows)
+
+  drop table a;
+  drop table b;
+  drop table c;
+  drop table d;
```
Comment (jaki) [inline src/postgres/src/test/regress/expected/yb_pg_join.out:4260]: ditto

File: src/postgres/src/test/regress/expected/yb_pg_join.out [hunk 17]
```diff
@@ -0,0 +1,6598 @@
...
+(1 row)
+
+SELECT b.* FROM b LEFT JOIN a ON (b.a_id = a.id) WHERE (a.id IS NULL OR a.id > 0);
+ id | a_id
+----+------
+  1 |
+(1 row)
+
+drop table a;
+drop table b;
```
Comment (jaki) [inline src/postgres/src/test/regress/expected/yb_pg_join.out:4280]: ditto

File: src/postgres/src/test/regress/expected/yb_pg_join.out [hunk 18]
```diff
@@ -0,0 +1,6598 @@
...
+$$
+  select count(*) from simple r join simple s using (id);
+$$);
+ initially_multibatch | increased_batches
+----------------------+-------------------
+                      |
+(1 row)
+
+-- parallel with parallel-oblivious hash join
+set local max_parallel_workers_per_gather = 2;
```
Comment (jaki) [inline src/postgres/src/test/regress/expected/yb_pg_join.out:5912]: Missing lines
```
rollback to settings;
```
and
```
savepoint settings;
```
If they don't work, still keep them but as commented out, and add TODO comments about them.

Ditto below.

File: src/postgres/src/test/regress/expected/yb_pg_join.out [hunk 19]
```diff
@@ -0,0 +1,6598 @@
...
+  select generate_series(1, 20000) AS id, 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa';
+--  ALTER TABLE SET not supported. See https://github.com/YugaByte/yugabyte-db/issues/1124
+--alter table simple set (parallel_workers = 2);
+-- Make a relation whose size we will under-estimate.  We want stats
+-- to say 1000 rows, but actually there are 20,000 rows.
+create table bigger_than_it_looks as
+  select generate_series(1, 20000) as id, 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa';
+--alter table bigger_than_it_looks set (autovacuum_enabled = 'false');
+--alter table bigger_than_it_looks set (parallel_workers = 2);
+update pg_class set reltuples = 1000 where relname = 'bigger_than_it_looks';
```
Comment (jaki) [inline src/postgres/src/test/regress/expected/yb_pg_join.out:5860]: Where's the line
```
analyze bigger_than_it_looks;
```
Same instructions as before.

File: src/postgres/src/test/regress/expected/yb_pg_join.out [hunk 20]
```diff
@@ -0,0 +1,6598 @@
...
+         Hash Cond: (r.id = s.id)
+         ->  Seq Scan on simple r
+         ->  Hash
+               ->  Seq Scan on simple s
+(6 rows)
+
+select count(*) from simple r join simple s using (id);
+ count
+-------
+     0
```
Comment (jaki) [inline src/postgres/src/test/regress/expected/yb_pg_join.out:5898]: Okay, this count is wrong.  Many others below are wrong.  I suggest trimming off the file from this section onwards.

File: src/postgres/src/test/regress/expected/yb_pg_join.out [hunk 21]
```diff
@@ -0,0 +1,6598 @@
...
+(0 rows)
+
+SELECT '' AS "xxx", *
+  FROM J1_TBL LEFT JOIN J2_TBL USING (i) WHERE (i = 1) order by 1, 2, 3, 4, 5;
+ xxx | i | j |  t  | k
+-----+---+---+-----+----
+     | 1 | 4 | one | -1
+(1 row)
+
+--
```
Comment (jaki) [inline src/postgres/src/test/regress/expected/yb_pg_join.out:1848]: Explain joins start to get different from here.  I think we can either trim the file at this point or keep the differing join results.  I lean towards keeping so that we can see when we improve the join cost estimates.

**Comment by jaki** [comment]
NOTE: this depends on pull request #5262 (test fails).

**Comment by jaki** [comment]
Now that PR #5262 is merged (commit 09d0f9c444dfd947af4d19d40e821e45927870d0), please merge or rebase latest master.

File: src/postgres/src/test/regress/expected/yb_pg_join.out (hunk 14 above)
Comment (ngov17) [reply src/postgres/src/test/regress/expected/yb_pg_join.out:2789]: I kept getting this error:` ERROR:  relation "a" already exists`. Same thing for tables b, c, and d.

Comment (jaki) [reply src/postgres/src/test/regress/expected/yb_pg_join.out:2789]: I see.  Please add TODO comments referencing issue #5313 for these.

**Comment by ngov17** [comment]
@jaki I've made the change in the latest commit:
1) All Style Issues
2) Removed all the code from the where `savepoint settings `is required. (Due to `rollback to settings` causing different results)

File: src/postgres/src/test/regress/yb_pg_join_schedule [hunk 22]
```diff
@@ -0,0 +1,22 @@
...
+####################################################################################################
+# Dependencies for `yb_pg_join`
+test: yb_pg_text
+test: yb_pg_int4
+test: yb_pg_int2
+test: yb_pg_int8
+test: yb_pg_float8
+
+# Dependencies for `yb_pg_join`
+#TODO: add yb_pg_inherit
```
Comment (jaki) [inline src/postgres/src/test/regress/yb_pg_join_schedule:14]:
```suggestion
# TODO: add yb_pg_inherit
```

**Comment by ngov17** [comment]
@jaki  Done in my latest commit.

**Comment by jaki** [comment]
Hi, @ngov17.  You forgot to check other schedules when renaming individual test files.  For example, you renamed `yb_text` to `yb_pg_text`, but some schedules still have `yb_text`.  One way to find these is `git grep yb_text src/postgres/src/test/regress/*_schedule`.  Don't forget the other renames besides `yb_text`.

This issue causes two Java tests to fail.  `TestPgForeignKey` is failing for known reasons, so that don't worry about that one.  Please find the other two and fix them.

**Comment by ngov17** [comment]
Hi @jaki. So these were the files that I renamed: yb_int2, yb_text, and yb_float8 to yb_pg_<name> respectively as they were ported tests. The two pg regress java tests that were failing due to not modifying the other schedules to reflect the change are: TypesString and TypesNumber. I have committed the changes and made sure these two tests work. Thanks!

**Comment by jaki** [comment]
Hi, @ngov17.  It looks like `TestPgRegressJoins` is flaky.  Some joins turn out differently, probably due to cost estimates being imprecise.  I'm not sure what the best way to go about this is.

One way is to cut out more of this test.  Most of the items in the test may not be useful because we don't really match vanilla postgres's expectations anyway.  I previously pointed out two potential cutoff points, and I suggested the later one, but the earlier one might be the easy way out at this point.

We could also dig into the flakiness.  Run `./yb_build.sh --java-test TestPgRegressJoins -n 200 -tp 1` and look at the logs for the failed tests.  Investigate each difference and either eliminate the (likely `EXPLAIN`) query or adjust it to be more deterministic.  This I imagine to be much more involved.

cc: @m-iancu

**Comment by ngov17** [comment]
@jaki I have cut out the test at the earlier point you mentioned in https://github.com/yugabyte/yugabyte-db/pull/5190#discussion_r463397441. Thanks!

File: src/postgres/src/test/regress/sql/yb_pg_join.sql [hunk 23]
```diff
@@ -0,0 +1,365 @@
...
+begin;
+set geqo = on;
+set geqo_threshold = 2;
+select count(*) from tenk1 x where
+  x.unique1 in (select a.f1 from int4_tbl a,float8_tbl b where a.f1=b.f1) and
+  x.unique1 = 0 and
+  x.unique1 in (select aa.f1 from int4_tbl aa,float8_tbl bb where aa.f1=bb.f1);
+rollback;
+
+-- yb_pg_join ends here. See: https://github.com/yugabyte/yugabyte-db/pull/5190#issuecomment-672361626
```
Comment (jaki) [inline src/postgres/src/test/regress/sql/yb_pg_join.sql:365]: newline at eof

Comment (ngov17) [reply src/postgres/src/test/regress/sql/yb_pg_join.sql]: @jaki Done in the latest commit. Thanks

**Comment by jaki** [comment]
I've been seeing test failures every once in a while.  Run `./yb_build.sh --java-test TestPgRegressJoin -n 1000`.

Most are on the second create table:

```diff
   1  --                                                                                                           1  --
   2  -- JOIN                                                                                                      2  -- JOIN
   3  -- Test JOIN clauses                                                                                         3  -- Test JOIN clauses
   4  --                                                                                                           4  --
   5  CREATE TABLE J1_TBL (                                                                                        5  CREATE TABLE J1_TBL (
   6    i integer,                                                                                                 6    i integer,
   7    j integer,                                                                                                 7    j integer,
   8    t text                                                                                                     8    t text
   9  );                                                                                                           9  );
  10  CREATE TABLE J2_TBL (                                                                                       10  CREATE TABLE J2_TBL (
  11    i integer,                                                                                                11    i integer,
  12    k integer                                                                                                 12    k integer
  13  );                                                                                                          13  );
                                                                                                              >   14  ERROR:  Operation failed. Try again.: Transaction aborted: a2fda428-582a-4e1f-8ae9-0f8af43050e5
```

One instance, I saw it on the following insert:

```
   1  --                                                                                                           1  --
   2  -- JOIN                                                                                                      2  -- JOIN
   3  -- Test JOIN clauses                                                                                         3  -- Test JOIN clauses
   4  --                                                                                                           4  --
   5  CREATE TABLE J1_TBL (                                                                                        5  CREATE TABLE J1_TBL (
   6    i integer,                                                                                                 6    i integer,
   7    j integer,                                                                                                 7    j integer,
   8    t text                                                                                                     8    t text
   9  );                                                                                                           9  );
  10  CREATE TABLE J2_TBL (                                                                                       10  CREATE TABLE J2_TBL (
  11    i integer,                                                                                                11    i integer,
  12    k integer                                                                                                 12    k integer
  13  );                                                                                                          13  );
  14  INSERT INTO J1_TBL VALUES (1, 4, 'one');                                                                    14  INSERT INTO J1_TBL VALUES (1, 4, 'one');
                                                                                                              >   15  ERROR:  Operation failed. Try again.: Transaction aborted: d4d822cf-8c9a-4844-9665-6458048a32ac
                                                                                                              >   16  LINE 1: INSERT INTO J1_TBL VALUES (1, 4, 'one');
                                                                                                              >   17
```

I haven't been able to reproduce these issues when running a schedule with a single test containing the two create tables, so the issue is probably tied to the previous tests in the schedule.

**Comment by ngov17** [comment]
@jaki So I was looking over the tests extensively, and one thing I realized was what you mentioned in: https://github.com/yugabyte/yugabyte-db/pull/5190#discussion_r460336350. The removal of the `DROP TABLE` for yb_pg_int2 and yb_pg_float8 might be causing this. To test this theory, I ran the yb_pg_float8 in a dummy schedule multiple times without the `DROP TABLE` and there were a few intermittent errors. To solve this issue, I decided to create a file `yb_pg_create_join` which essentially just creates and populates the tables required for yb_pg_join from yb_pg_float8 and yb_pg_int2. Upon doing this and updating the schedule for join (by removing the tests for float8 and int2 so it doesn't affect any other tests), I ran the join test multiple times overnight and it seemed to be fine. I will continue to do more tests throughout the week to see if this is the problem or something else potentially. Thanks!

File: src/postgres/src/test/regress/yb_pg_join_schedule [hunk 24]
```diff
@@ -6,9 +6,8 @@
 # Dependencies for `yb_pg_join`
 test: yb_pg_text
 test: yb_pg_int4
-test: yb_pg_int2
 test: yb_pg_int8
-test: yb_pg_float8
+test: yb_pg_create_join
```
Comment (jaki) [inline src/postgres/src/test/regress/yb_pg_join_schedule:10]: `create_join` is not in `serial_schedule`.  Plus, it looks like some workaround.  Is this WIP and you need to now add back `yb_pg_int2` and `yb_pg_float8` since you added `DROP TABLE` back?

Comment (jaki) [reply src/postgres/src/test/regress/yb_pg_join_schedule:10]: To clarify, either change it to `yb_create_join` since it's not an original pg test OR (preferrably) remove it and bring back `yb_pg_int2` and `yb_pg_float8` if you can.

**Comment by ngov17** [comment]
@jaki Thanks for the feedback! My bad, I renamed the file to yb_create_join as it is not ported from postgres. So I removed the `DROP TABLE` for int2 and float8 and for some reason this drop table was added in the original yb files for int2 and float8 (if you open int2.sql or float8.sql, which are the original vanilla postgres files, this `DROP TABLE` is not present). Right now I am working on the theory that this removal of DROP TABLE might be the source of the issues we see in repeated tests. 

Based on that,  I added back those DROP TABLE as described in https://github.com/yugabyte/yugabyte-db/pull/5190#issuecomment-681105868 . However in doing so I cannot use the original `yb_pg_int2` or `yb_pg_float8` as dependencies for the test as we require the tables to remain in existence for `yb_pg_join`. So I created the file `yb_create_join` (which was formerly `yb_pg_create_join` to jut create these tables for `yb_pg_join`. 

Now, I can still add back the tests for `yb_pg_float8` and `yb_pg_int2` as dependencies to test that the populated tables work properly. However, the tables will immediately be deleted (due to the DROP) and only the ones from `yb_create_join` will remain and be used in join.

**Comment by jaki** [comment]
Oh, got it.

Anyway, I found that it still fails on the second CREATE TABLE (after rep 51 for me), so the drop table doesn't help.  Let's just bring things back to the way they were (no DROP TABLE and no create_join), merge this PR, and track this test failure issue separately.

**Comment by ngov17** [comment]
@jaki Thanks for the update! I reversed the commit to what it was (to commit  94deb45) before I enacted all the changes with the float8 and int2.

**Comment by jaki** [comment]
Hi @ngov17.  I'm not sure what you did, but you reverted the wrong commits.  Please revert the following commits:
1. 92dcbf7
1. bca3feb
1. 29cbafc
1. 890386f
1. 9d6872c

**Comment by ngov17** [comment]
@jaki I used `git revert -m 1 c1c38ae` to revert back to right before I merged this branch (master) with upstream (to update the repository with upstream, the official YB), basically reverting this branch to right before commit c1c38ae wherein I merged the branch with upstream). Once I did that, I used `git revert --no-commit 94deb45 ` which essentially directly reverses all of the remaining commits you mentioned in https://github.com/yugabyte/yugabyte-db/pull/5190#issuecomment-683176292 to commit 94deb45, which skips a prompt to add each consecutive commit revert as a separate commit, doing all these specific reverts you mentioned. So the current state of my file is as per https://github.com/yugabyte/yugabyte-db/pull/5190/commits/94deb455b3e441300f1dc1e6f3ad491819f2e2f5, which is exactly the last commit I was at before all the float8, int2 changes. 

I think the merge conflicts might be arising because I reversed my merge with upstream master, essentially changing all the upstream changes which won't be reflected in the current commit. Since I reversed my commit to a commit made over a week ago, I imagine there would be a few merge conflicts arising when merging to upstream/master.

**Comment by ngov17** [comment]
@jaki I'll try resolving the conflicts now and put another commit when that is done. Thanks for the feedback!

**Comment by ngov17** [comment]
@jaki I fixed the wrong reverts and there are no more merge conflicts with the base branch now! Thanks.

**Comment by jaki** [comment]
Commit 558da87 (Add new files) looks suspicious.  Submodule changes shouldn't be a part of this PR.  Please revert it.
Commit e439d57 is fine.

**Comment by ngov17** [comment]
@jaki Oops, looks like the submodules were added after my first revert and I just added it. I deleted the submodules and pushed it. Thanks for the feedback!

**Review by jaki — APPROVED** [review]
Thanks, @ngov17.
